pkg/storagev2/migration/migration.go

```diff
@@ -0,0 +1,111 @@
+package migration
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"sort"
+)
+
+type (
+	// StepFn is a function that migrates the storage to the next version
+	StepFn func(storage.Store) error
```

Comment:
I'd consider renaming `StepFn` to `Action` and `StepsMap` to `Steps`.
@aloknerurkar @vladopajic what do you think?

Reply:
If we zoom out and look for the most generic name for `func(storage.Store) error` what would it be? 

---

In the time of writing proposal I haven't got any good generic name idea for such function, because signature is vary broad and can fit many concepts. 

Decision was to have have highly specific name in the context of migration. At that time options like `step`, `increment`, `stage` were floating in my mind. 

Name `action` seems to be going in generic name direction. I would rather avoid generic name unless abstraction for this type is occurring in other places.

---

One potential name in generic category was `mutator` because it implies that `store` is going to be mutated in this function (bonus it's also a verb).

Reply:
I wouldn't say it's generic, you have to read it in the context of the package, which is `migration.Action` and `migration.Steps`. I think compared to `migration.StepFn` and `migration.StepsMap` they are clearer, especially in the case of `StepsMap` where I consider the `Map` suffix to be an unnecessary detail.

Reply:
`migration.Steps` is better then `migration.StepsMap` we agree on that. But for `migration.Action` I am not quite sure...

Reply:
We can go with `migration.StepFn`.